src/librustdoc/formats/cache.rs

```diff
@@ -156,10 +156,16 @@ impl Cache {
         // FIXME: this part is specific to HTML so it'd be nice to remove it from the common code
         for &(n, ref e) in &krate.externs {
             let src_root = match e.src {
-                FileName::Real(ref p) => match p.local_path().parent() {
-                    Some(p) => p.to_path_buf(),
-                    None => PathBuf::new(),
-                },
+                FileName::Real(ref p) => {
+                    if let Some(local_path) = p.local_path() {
+                        match local_path.parent() {
+                            Some(p) => p.to_path_buf(),
+                            None => PathBuf::new(),
+                        }
+                    } else {
+                        PathBuf::new()
+                    }
```

Comment:
This will always be `None`, because extern crates are always non-local. That makes me a very nervous - I think that will break `src_href` for any remapped path from another crate. Is there no other way to get the filename? It's ok if it's only a relative path, `clean_path` will strip `src_root` anyway.